py-polars/polars/testing/asserts.py

```diff
@@ -77,11 +86,21 @@ def assert_frame_equal(
             f"Columns {right_not_left} in right frame, but not in left"
         )
 
-    if check_column_names:
-        if left.columns != right.columns:
-            raise AssertionError("Columns are not in the same order")
+    if check_column_order and left.columns != right.columns:
+        raise AssertionError(
+            f"Columns are not in the same order:\n{left.columns!r}\n{right.columns!r}"
+        )
+
+    if not check_row_order:
+        try:
+            left = left.sort(by=left.columns)
+            right = right.sort(by=right.columns)
```

Comment:
What if the columns are in different order and `check_column_order=False`?

In that case, you should sort using the same order, i.e. both by the order in the left dataframe, or both by the order in the right dataframe. Otherwise different column ordering will affect your sort.

Reply:
Good catch :)

Reply:
Fixed (and tweaked the associated unit test to fail if it wasn't).